### ISSUE TITLE
Feature request : Update Firware

### ISSUE THREAD

**Issue by guillaumemolter** [tomasf/microprint#14]
Hey @tomasf,
Thanks for the amazing software.

When I start Microprint it tells me that my Firmware is untested (probably because I have an old firmware and haven't been able to launch the M3D software in weeks).

How hard would it be for you to implement an update firmware feature?

Thanks

**Comment by guillaumemolter** [comment]
I should specify that so far I didn't have any problems printing. But I would rather use the latest firmware.

**Comment by tomasf** [comment]
Firmware updating is definitely on the roadmap. It's sort of the last remaining thing you need M3D's software for. I haven't attempted it yet, mostly because it's a somewhat convoluted process of dropping back into bootloader mode for a while. There's also the question of how to obtain the actual firmware data. It's embedded in the M3D program as an MSIL manifest resource. Extracting it programmatically is certainly possible, but a bit tricky, and would require that you at least download M3D's software first. Embedding it into MicroPrint would be easier, but that's a bit of a violation of their copyright. The most awesome way for users would be if MicroPrint downloaded M3D's software itself and did all the extraction work, but that's a big thing to implement, obviously. :-)

**Comment by guillaumemolter** [comment]
Yeah I figured getting the Firmware would actually be the hard part. I think the M3D has a good community (and not thanks to M3D) and while I understand it's borderline "illegal" to extract the Firmware from their app if you give us a way to import it in the M3D I'm sure someone will extract it and make it available to the community.
